src/WebJobs.Script/Workers/SharedMemoryDataTransfer/SharedMemoryMap.cs

```diff
@@ -0,0 +1,599 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.IO.MemoryMappedFiles;
+using System.Runtime.InteropServices;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Script.Workers.SharedMemoryDataTransfer
+{
```

Comment:
As discussed, Create a base abstract class and create OS specific implementations